### PR TITLE
Add Global NCC

### LINKED ISSUE #512: Add image dissimilarity losses: global NCC, L1, TukeyBiweight
## Subject of the feature

What are you trying to do and how would you want to do it differently?
From issue #424 
@acasamitjana  suggested adding new losses to the project.
"Mutual information (specially to compare with other intermodal registrations methods), Tukey Biweight (a more robust SSIM metric)"

Is it something you currently you cannot do?
Yes

Is this related to an issue/problem?
No.

Has the feature been requested before? If yes, please provide a link to the issue.
No.

If the feature request is approved, would you be willing to submit a PR? 

M.

### PR DESCRIPTION
**Pull request by acasamitjana**
# Description

In some contexts, the global NCC may be more useful than the local NCC. 

Fixes #512

## Type of change

What types of changes does your code introduce to DeepReg?

_Please check the boxes that apply after submitting the pull request._

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation Update (fix or improvement on the documentation)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (if none of the other choices apply)

## Checklist

_Please check the boxes that apply after submitting the pull request._

_If you're unsure about any of them, don't hesitate to ask. We're here to help! This is
simply a reminder of what we are going to look for before merging your code._

- [X] I have
      [installed pre-commit](https://deepreg.readthedocs.io/en/latest/contributing/setup.html)
      using `pre-commit install` and formatted all changed files. If you are not
      certain, run `pre-commit run --all-files`.
- [X] My commits' message styles matches
      [our requested structure](https://deepreg.readthedocs.io/en/latest/contributing/commit.html),
      e.g. `Issue #<issue number>: detailed message`.
- [X] I have updated the
      [change log file](https://github.com/DeepRegNet/DeepReg/blob/main/CHANGELOG.md)
      regarding my changes.
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by acasamitjana** [comment]
Hi @mathpluscode a really easy merge that closes issue #512 . There are other losses that, if needed, we'll implement later on. @NMontanaBrown can you take a quick look as well?

File: test/unit/test_loss_image.py [hunk 1]
```diff
@@ -118,3 +118,33 @@ def test_get_config(self):
...
+        [
+            (0.6, 0.3, (3,), 1),
+            (0.6, 0.3, (3, 3), 1),
+            (0.6, -0.3, (3, 3), 1),
+            (0.6, 0.3, (3, 3, 3), 1),
+        ],
+    )
+    def test_output(self, y_true, y_pred, shape, expected):
+
+        y_true = y_true * np.zeros(shape=shape)
```
Comment (NMontanaBrown) [inline test/unit/test_loss_image.py:135]: What's the point of this op, if we do y_true * np.zeros == np.zeros? IE what is the point of reassigning to np.zeros from pytest parametrize? Not sure if this is a bug. Can you explain further?

**Review by NMontanaBrown — COMMENTED** [review]
Left a comment about the unit test.

File: test/unit/test_loss_image.py (hunk 1 above)
Comment (acasamitjana) [reply test/unit/test_loss_image.py:135]: Yes sorry, my miss.

File: test/unit/test_loss_image.py [hunk 2]
```diff
@@ -118,3 +118,32 @@ def test_get_config(self):
...
+        [
+            (0.6, 0.3, (3, 3), 1),
+            (0.6, 0.3, (3, 3, 3), 1),
+            (0.6, -0.3, (3, 3, 3), 1),
+            (0.6, 0.3, (3, 3, 3, 3), 1),
+        ],
+    )
+    def test_output(self, y_true, y_pred, shape, expected):
+
+        y_true = y_true * np.ones(shape=shape)
```
Comment (mathpluscode) [inline test/unit/test_loss_image.py:135]: I suggest to use tf.ones to be safe, as `np.ones` tend to define numpy arrays in float64 :)

**Comment by acasamitjana** [comment]
Thanks for your comments. I have updated the requested changes and merged the master branch.

**Review by NMontanaBrown — APPROVED** [review]

**Review by mathpluscode — APPROVED** [review]

**Pull request merged**